### ISSUE TITLE
Loadbalancer: cant set SourceIP / Port anymore

### ISSUE THREAD

**Issue by toughIQ** [rancher/rancher#6727]
**Rancher Version:**
v1.2.0-pre4-rc11
**Docker Version:**
1.12.2
**OS and where are the hosts located? (cloud, bare metal, etc):**
RHEL 7.2 (VMware)
**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single Node, external MariaDB
**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle
**Steps to Reproduce:**
Upgrading from rc6 to rc11. It told that Loadbalancers have to be upgraded.
**Results:** 
Upgrade Failed.
**Expected:**
Smooth LB upgrade.

I suspect: I had a LB setting, where I added a dedicated IP to bind to. I tried to clone my old LB and configure my requirements with the new one. I couldnt find any option/possibility to add a __SourceIP/Port__ setting. Seems this whole section, which is there in v1.1.4, is missing from v1.2.

Am I missing something?

**Comment by deniseschannon** [comment]
Upgrades will only be supported from 1.1.4 to 1.2.0 and 1.2.0-pre3 to 1.2.0. We've gone from v1 Load balancer to a v2 load balancers and in 1.2.0, we will be converting old v1 LBs to the new v2 LBs. 

There is an extensive documentation update regarding the new Load balancers that will be coming with 1.2.0. 

Can you share your docker-compose.yml of your old LB?

**Comment by toughIQ** [comment]
I am sorry, but I nuked my old dev environment with this upgrade, so I dont have the old compose file anymore. Here is the current one, which doesnt start.
(IP address and domains randomized) 
```
  loadbalancer:
    ports:
    - 123.123.12.12:443:443/tcp
    labels:
      io.rancher.scheduler.global: 'true'
      io.rancher.loadbalancer.ssl.ports: '443'
      io.rancher.container.create_agent: 'true'
      io.rancher.container.agent.role: environmentAdmin
      io.rancher.loadbalancer.target.portus: portus.mydomain.at:443=3000
      io.rancher.loadbalancer.target.registry: registry.mydomain.at:443=5000
    tty: true
    image: rancher/lb-service-haproxy:v0.4.1
    links:
    - portus:portus
    - registry:registry
    stdin_open: true
```
But even if the automatic upgrade doesnt work, I cant recreate my old config using the new interface. There seems to be no place to put the IP to listen to.

**Comment by amit213** [comment]
This is a major issue and pain point encountered since we upgraded to rancher server v1.2. 

The "add load balancer" option via UI - used to previously allow the users to bind to specific IP addresses along with the port. This capability was vital.  It allowed for a more secure design, where we have external facing hosts with dual IPs (one external, other internal with rancher agent running on it).  Adding load balancer with specific IP:port allowed us to control the behavior of haproxy. 

That is broken. Please share guidance asap on how that functionality can be achieved in v1.2 server.

**Comment by vincent99** [comment]
Extensive discussion in https://github.com/rancher/rancher/issues/6852 .  TL;DR: The functionality still exists but is not in the UI because it does far more harm than good for most users with hosts that have one public IP.  But we will put it back as a disabled-by-default option in a future release.

@amit213 Are you sure you don't just want load balancers with listeners set to Internal access?  That makes it not publish a port to the host at all, but the balancer is still available to any other container by DNS/it's 10.42.x.y IP.  You'd only need "public" + binding to an internal rfc1918 IP if there are other things external to Rancher that need to connect to it.

**Comment by amit213** [comment]
@vincent99 For most part we are okay with load balancer service listening onto the internal NICs / network. However, specific topologies warrant external entities (like GSLBs) to connect to external IP where a rancher exposed load balancer service is running. The rancher loadbalancer service would have to be bound to public IP, and then fan out the incoming request to private IPs / containers that serve those requests. 

Thanks for point to #6852 . Will check it out. Look forward to you adding back as "disabled-by-default" in your near term release.
